Source/Serilog.Enrichers.Span/ActivityEnricher.cs

```diff
@@ -9,6 +9,60 @@ namespace Serilog.Enrichers.Span
     /// </summary>
     public class ActivityEnricher : ILogEventEnricher
     {
+#if NET5_0
+        /// <summary>
+        /// Enrich the log event.
+        /// </summary>
+        /// <param name="logEvent">The log event to enrich.</param>
+        /// <param name="propertyFactory">Factory for creating new properties to add to the event.</param>
+        public void Enrich(LogEvent logEvent, ILogEventPropertyFactory propertyFactory)
+        {
+            var activity = Activity.Current;
+
+            if (activity is not null)
+            {
+                AddSpanId(logEvent, activity);
+                AddTraceId(logEvent, activity);
+                AddParentId(logEvent, activity);
+            }
+        }
+
+        private static void AddSpanId(LogEvent logEvent, Activity activity)
+        {
+            var property = activity.GetCustomProperty("Serilog.SpanId");
+            if (property is null || property is not LogEventProperty logEventProperty)
+            {
+                logEventProperty = new LogEventProperty("SpanId", new ScalarValue(activity.GetSpanId()));
+                activity.SetCustomProperty("Serilog.SpanId", logEventProperty);
```

Comment:
I'm not familiar with `GetCustomProperty` and `SetCustomProperty` and the docs seem useless. I assume it's just a bag of stuff you can store things in and has no impact on Open Telemetry itself.

Reply:
I haven't used `GetCustomProperty` and `SetCustomProperty` in the past, but I also assume that it dose not have any impact on anything.

Reply:
Is there any way to make sure?

Reply:
If not, I'll just leave a StackOverflow question just to make sure.

Reply:
> Allow attaching any object to the Activity object through the Set/GetCustomProperty for the sake of the extendability without manually sub-classing Activity or maintaining a dictionary mapping activity to the other objects. Examples of that, OT implementation still depends on Activity and maintains a table mapping from Activity to Span. Another example is some teams were working around the lack of missing Links and Contexts features and manually doing extra work to include these features. Although we are adding the Links and Contexts, we can get more future scenarios to extend Activity (e.g. attaching the status codes/exceptions).

From https://github.com/dotnet/designs/blob/main/accepted/2020/diagnostics/activity-improvements.md

It seems what we are doing is ok. We could tag a dotnet team member and ask them just to be sure.

Reply:
Perfect